### PR TITLE
Oldosxbuild

### PR DESCRIPTION
**Pull request by MrChico**
This was needed in order to be able to build on macos High Sierra. 
Somewhat mysteriously, the `OSX_DEPLOYMENT_TARGET` is set to higher than what I'm on (High Sierra is `10.13`), but at least others seem to share this experience https://stackoverflow.com/questions/52310835/xcode-10-call-to-unavailable-function-stdvisit

### CODE REVIEW

**Comment by MrChico** [comment]
solves https://github.com/ethereum/solidity/issues/8506

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,6 +11,7 @@ eth_policy()
 
 # project name and version should be set after cmake_policy CMP0048
 set(PROJECT_VERSION "0.6.5")
+set(CMAKE_OSX_DEPLOYMENT_TARGET "10.14")
```
Comment (leonardoalt) [inline CMakeLists.txt:15]: I'm not sure this would be the right fix... then we'd need to bump the OSX version every now and then

Comment (MrChico) [reply CMakeLists.txt:15]: This is a minimal target though... wouldn't bumping it be appropriate when the repo becomes incompatible with older os?

Comment (leonardoalt) [reply CMakeLists.txt:15]: Hmm yea, that makes sense. I'm wondering *why* it is incompatible though

Comment (leonardoalt) [reply CMakeLists.txt:15]: Ok, apparently because of OSX's `libc`, `std::visit` does require 10.14 regardless the compiler version. Weird, but I guess then this is the right fix

Comment (axic) [reply CMakeLists.txt:15]: Can you leave a comment above this `set()` stating it is needed for `std::visit`?

File: scripts/install_deps.sh [hunk 2]
```diff
@@ -105,6 +105,9 @@ case $(uname -s) in
         brew update
         brew install boost
         brew install cmake
+        brew install cln
+        brew install z3
+        brew install cvc4
```
Comment (axic) [inline scripts/install_deps.sh]: These don't seem to be related to the PR?

Comment (MrChico) [reply scripts/install_deps.sh]: Well, they are needed for installation on macos. Would you prefer a separate PR?

Comment (leonardoalt) [reply scripts/install_deps.sh]: They are generally optional dependencies, it should be possible to compile without them. Does that not work there?

Comment (MrChico) [reply scripts/install_deps.sh]: It complained at least that cln was missing. I have the other two installed already. Could it be that cln is needed if z3 or cvc4 are present?

Comment (leonardoalt) [reply scripts/install_deps.sh]: Exactly, cln is needed by cvc4

Comment (MrChico) [reply scripts/install_deps.sh]: Things seem a little inconsistent across this script. `cvc4` is installed for Arch linux, but no other distro. `z3` installed for Debian and Ubuntu.

Comment (leonardoalt) [reply scripts/install_deps.sh]: Right, this is actually wrong I think

Comment (MrChico) [reply scripts/install_deps.sh]: So you prefer these lines left out?

Comment (leonardoalt) [reply scripts/install_deps.sh]: Yep

**Comment by MrChico** [comment]
Oki, should be all good now then

**Review by axic — APPROVED** [review]
